### DISCUSSION TITLE
Gate.io futures

### DISCUSSION THREAD

**Discussion by ibmua** [bmoscon/cryptofeed#583] [Q&A]
Can I get data for gateio futures with cryptofeed?

**Comment by ibmua** [comment]
Simply using as

```
symbol = Symbol('FLOW', 'USDT', type=PERPETUAL)
fh.add_feed(Gateio(symbols=[symbol], channels=[L2_BOOK], callbacks={L2_BOOK: book}))
```

Doesn't work.

**Comment by bmoscon** [comment]
you can tell what symbols an exchange supports by looking at the output of the `symbols` method. I dont think it supports futures yet

**Reply by ibmua** [reply]
Thanks! Yes, indeed 
`
print( Gateio.symbols() )
`
outputs available trading pairs, and those are indeed only from spot market.
